gcc_deployment_script.sh

```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
+
+set -e
+
+usage() {
+  cat <<EOF
+  usage: $0
+  mandatory options:  -a ARTIFACTS_DIR
+                      -d DEPLOYMENT_VERSION
+                      -r REPOSITORY_NAME
+                      -g GIT_REF
+                      -u REMOTE_USER
+                      -i REMOTE_IP
+                      -t REMOTE_DEST_DIR
+  This script creates an appropriately named directory
+  with artifacts and deploys it to a provided deployment
+  server.
+EOF
+}
+
+send_dir() {
+  scp_src="$deployment_dir"
+  remote_host="$remote_user@$remote_ip"
+  remote_dest="$remote_dest_dir/$repository_name/$deployment_version"
+  scp_dest="$remote_host:$remote_dest"
+
+  echo -e "\n- Sending dir to deployment server"
+  echo -e "  src: $scp_src"
+  echo -e "  dest: $scp_dest"
+  ssh "$remote_host" "mkdir -p $remote_dest"
+  scp -r "$scp_src" "$scp_dest"
```

Comment:
I personally think it is better if you create the directory tree in the VM you are building and them copy using scp, instead of recreating the directory tree using 2 ssh commands

Reply:
Done.